src/content/1.7/modules/sample_modules/view-order-hooks-demo.md

```diff
@@ -0,0 +1,721 @@
+    {
+        return $this->orderId;
+    }
+
+    public function getReference(): string
+    {
+        return $this->reference;
+    }
+}
+```
+
+Orders collection `Orders.php` in `src/Collection`:
+
+```php
+<?php
+declare(strict_types=1);
+
+namespace PrestaShop\Module\DemoViewOrderHooks\Collection;
+
+use PrestaShop\Module\DemoViewOrderHooks\DTO\Order;
+use PrestaShop\PrestaShop\Core\Data\AbstractTypedCollection;
+
+final class Orders extends AbstractTypedCollection
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function getType()
+    {
+        return Order::class;
+    }
+}
+```
+
+`OrderRepository` in `src\Repository`:
+
+```php
+<?php
+declare(strict_types=1);
+
+namespace PrestaShop\Module\DemoViewOrderHooks\Repository;
+
+use DateTimeImmutable;
+use Db;
+use Order as PrestaShopOrder;
+use PrestaShop\Module\DemoViewOrderHooks\Collection\Orders;
+use PrestaShop\Module\DemoViewOrderHooks\DTO\Order;
+
+class OrderRepository
+{
+    /**
+     * @var Db
+     */
+    private $db;
+
+    public function __construct()
+    {
+        $this->db = Db::getInstance();
+    }
+
+    /**
+     * Get all orders that a customer has placed.
+     */
+    public function getCustomerOrders(int $customerId, array $excludeOrderIds = []): Orders
+    {
+        $orders = PrestaShopOrder::getCustomerOrders($customerId);
+        $ordersCollection = new Orders();
+
+        foreach ($orders as $order) {
+            if (in_array($order['id_order'], $excludeOrderIds)) {
+                continue;
+            }
+
+            $ordersCollection->add(new Order(
+                (int) $order['id_order'],
+                $order['reference'],
+                (int) $order['current_state'],
+                new DateTimeImmutable($order['date_add'])
+            ));
+        }
+
+        return $ordersCollection;
+    }
+}
+
+```
+
+Let's create `SignaturePresenter` class responsible for returning order customer data  
+in `src/Presenter/`:
+
+```php
+<?php
+declare(strict_types=1);
+
+namespace PrestaShop\Module\DemoViewOrderHooks\Presenter;
+
+use Gender;
+use Order;
+use PrestaShop\Module\DemoViewOrderHooks\Entity\Signature;
+
+class SignaturePresenter
+{
+    /**
+     * @var string
+     */
+    private $signatureImgDir;
+
+    public function __construct(string $signatureImgDir)
+    {
+        $this->signatureImgDir = $signatureImgDir;
+    }
+
+    public function present(Signature $signature, int $languageId): array
+    {
+        $order = new Order($signature->getOrderId());
+        $customer = $order->getCustomer();
+        $gender = new Gender($customer->id_gender, $languageId);
+
+        return [
+            'firstName' => $customer->firstname,
+            'lastName' => $customer->lastname,
+            'gender' => $gender->name,
+            'filename' => $this->signatureImgDir.$signature->getFilename()
+        ];
+    }
+}
+```
+
+Then lets create services configuration for the above classes in `config/services.yml`. 
```

Comment:
Please explain briefly why services architecture is good for maintainability and how Symfony Dependency Injection, which uses our YAML config file here, is a good solution to implement it